pandas/tests/extension/test_arrow.py

```diff
@@ -664,6 +664,14 @@ def test_EA_types(self, engine, data, request):
                     reason=f"Parameterized types with tz={pa_dtype.tz} not supported.",
                 )
             )
+        elif pa.types.is_timestamp(pa_dtype):
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    raises=ValueError,
+                    strict=False,
+                    reason="https://github.com/pandas-dev/pandas/issues/49767",
+                )
+            )
```

Comment:
@WillAyd here's an issue that's come up - if a Series of if `timestamp[ns][pyarrow]` dtype, then `to_csv` won't print everything with the same precision. For example, it might print:
- `2020-01-01 01:01:01.000001` (if nanoseconds are present)
- `1999-01-01 00:00:00` (if there are a whole number of seconds)

But then, the format will be guessed from the first row to be `%Y-%m-%d %H:%M:%S.%f`, and parsing the second row will fail because it doesn't match that format.

Reckon it's OK to xfail this test for now, or is it a blocker?

Note that this isn't an issue for `datetime[64]` dtype, because then they get printed as:
- `2020-01-01 01:01:01.000001`
- `1999-01-01 00:00:00.000000`
and both conform to the `%Y-%m-%d %H:%M:%S.%f` format

Reply:
It's because the test
```
pandas/tests/extension/test_arrow.py::TestBaseParsing::test_EA_types[timestamp[s]-c]
```
has got all inputs without nanoseconds, so they get printed in the same format and the test passes, whereas
```
pandas/tests/extension/test_arrow.py::TestBaseParsing::test_EA_types[timestamp[us]-c]
```
has got some inputs with nanoseconds and others without, so they get printed in different formats and the test fails.

Perhaps better to make the condition stricter and only xfail one of them, thanks for having brought this up